datashuttle/utils/formatting.py

```diff
@@ -216,26 +216,28 @@ def update_names_with_datetime(names: List[str]) -> None:
     Format using key-value pair for bids, i.e. date-20221223_time-
     """
     date = str(datetime.datetime.now().date().strftime("%Y%m%d"))
-    format_date = f"date-{date}"
+    date_with_key = f"date-{date}"
 
     time_ = datetime.datetime.now().time().strftime("%H%M%S")
-    format_time = f"time-{time_}"
+    time_with_key = f"time-{time_}"
+
+    datetime_with_key = f"datetime-{date}T{time_}"
 
     for i, name in enumerate(names):
-        if tags("datetime") in name:  # must come first
+        # datetime conditional must come first.
+        if tags("datetime") in name:
             name = add_underscore_before_after_if_not_there(
                 name, tags("datetime")
             )
-            datetime_ = f"{format_date}_{format_time}"
-            names[i] = name.replace(tags("datetime"), datetime_)
+            names[i] = name.replace(tags("datetime"), datetime_with_key)
 
         elif tags("date") in name:
             name = add_underscore_before_after_if_not_there(name, tags("date"))
-            names[i] = name.replace(tags("date"), format_date)
```

Comment:
`format_date` included the `date-` key, this was super-unclear from the variable name ('format' soundsl ike the date / time have been formatted correctly but actually meant the BIDS key had been added) so have updated these and switched the `datetime` format to use the date / time only.